app/src/main/res/layout/current_weather_layout.xml

```diff
@@ -0,0 +1,135 @@
+                android:layout_marginBottom="8dp"
+                android:text="@string/current_temp"
+                android:textSize="28sp"
+                android:textStyle="bold"
+                app:layout_constraintBottom_toTopOf="@id/current_location_text"
+                app:layout_constraintEnd_toStartOf="@id/barrier2"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/current_weather_type_text" />
+
+            <TextView
+                android:id="@+id/current_location_text"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="8dp"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="8dp"
+                android:layout_marginBottom="8dp"
+                android:text="@string/current_location"
+                android:textSize="18sp"
+                app:layout_constraintBottom_toTopOf="@id/current_date_text"
+                app:layout_constraintEnd_toStartOf="@id/barrier2"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/current_temp_text" />
+
+            <TextView
+                android:id="@+id/current_date_text"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="8dp"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="8dp"
+                android:text="@string/current_date"
+                android:textSize="18sp"
+                app:layout_constraintBottom_toBottomOf="@+id/current_weather_image_view"
+                app:layout_constraintEnd_toStartOf="@+id/barrier2"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@+id/current_location_text"
+                app:layout_constraintVertical_bias="1.0" />
+
+            <ImageView
+                android:id="@+id/current_weather_image_view"
+                android:layout_width="200dp"
+                android:layout_height="200dp"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="8dp"
+                android:layout_marginBottom="8dp"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toEndOf="@id/barrier2"
+                app:layout_constraintTop_toBottomOf="@id/guideline"
+                app:layout_constraintVertical_bias="0.5"
+                app:srcCompat="@drawable/sunny_image" />
+
+            <androidx.constraintlayout.widget.Barrier
```

Comment:
For now, it's okay, but you can achieve almost this layout by using just one guideline. Always try to keep xml files as lean as possible.

Reply:
Should i leave it for now or change it?

Reply:
Keep it as is for now.